ecl/hql/hqlgram2.cpp

```diff
@@ -12045,7 +12045,7 @@ IHqlExpression * HqlGram::createCheckMatchAttr(attribute & attr, type_t tc)
         type.setown(makeUtf8Type(UNKNOWN_LENGTH, NULL));
         break;
     default:
-        throwUnexpectedType(type);
```

Comment:
This is not a desirable change. What was the error that provoked it?

Reply:
@richardkchapman 
[ 56%] Built target hpccmetrics_prometheussink
In file included from /mnt/e/HPCC/HPCC-Platform/ecl/hql/hqlgram2.cpp:24:
/mnt/e/HPCC/HPCC-Platform/ecl/hql/hqlgram2.cpp: In member function ‘IHqlExpression* HqlGram::createCheckMatchAttr(attribute&, type_t)’:
/mnt/e/HPCC/HPCC-Platform/ecl/hql/hqlerrors.hpp:621:62: error: ‘this’ pointer is null [-Werror=nonnull]
  621 | #define throwUnexpectedType(type)   throw MakeStringException(ECODETEXT(HQLERR_UnexpectedType), type->queryTypeName(), sanitizeSourceFile(__FILE__), __LINE__)
/mnt/e/HPCC/HPCC-Platform/ecl/hql/hqlgram2.cpp:12048:9: note: in expansion of macro ‘throwUnexpectedType’
12048 |         throwUnexpectedType(type);
      |         ^~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [ecl/hql/CMakeFiles/hql.dir/build.make:256: ecl/hql/CMakeFiles/hql.dir/hqlgram2.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:11105: ecl/hql/CMakeFiles/hql.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
russ@DESKTOP-58E0PSA:/mnt/e/HPCC/build$

Reply:
I'm happy enough with that change - I don't think it can ever occur - so throwUnexpected() is appropriate.